sdk/go/property/map.go

```diff
@@ -105,6 +105,15 @@ func (m Map) Set(key string, value Value) Map {
 	return Map{cp}
 }
 
+// Delete produces a new map identical to the receiver with given keys removed.
+func (m Map) Delete(keys ...string) Map {
+	cp := copyMapNonNil(m.m)
```

Comment:
It's OK to `delete` on a nil map, so we can do `copyMapMaybeNil`, This should save us an allocation when deleting from an empty map.
```suggestion
	cp := copyMapMaybeNil(m.m)
```